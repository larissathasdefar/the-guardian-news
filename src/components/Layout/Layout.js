import React from 'react'
import _ from 'prop-types'
import styled from 'styled-components'
import Snackbar from '../Snackbar'
import Typography from '@material-ui/core/Typography'

const Wrapper = styled.div`
    a {
        text-decoration: none;
    }
`

const Header = styled.div`
    display: flex;
    height: calc(100px - 32px);
    background: #e7edef;
    align-items: center;
    padding: 16px;
`

const Body = styled.div`
    min-height: calc(100vh - 100px);
    max-width: 880px;
    margin: 0 auto;
    padding: 10px;
`

const Layout = ({children, snack, onHideNotification, ...otherProps}) => {
    return (
        <Wrapper data-testid="layout-content">
            <Header {...otherProps}>
                <Typography variant="h3">The Guardian</Typography>
            </Header>
            <Body>{children}</Body>
            <Snackbar
                open={snack.open}
                variant={snack.type}
                onClose={onHideNotification}
                message={snack.message}
            />
        </Wrapper>
    )
}

Layout.propTypes = {
    snack: _.shape({
        open: _.bool,
        type: _.oneOf(['error', 'success']),
        message: _.string,
    }),
    children: _.node.isRequired,
    onHideNotification: _.func.isRequired,
}

Layout.defaultProps = {
    snack: {
        open: false,
        type: 'success',
        message: '',
    }
}

export default Layout
