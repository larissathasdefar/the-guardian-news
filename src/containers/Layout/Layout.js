import React from 'react'
import _ from 'prop-types'
import { connect } from 'react-redux'
import { hideNotification } from '../../actions/notification'
import Wrapper from '../../components/Layout'

const LayoutContainer = ({ children, ...otherProps }) => (
    <Wrapper {...otherProps}>
        {children}
    </Wrapper>
)

LayoutContainer.propTypes = {
    children: _.node.isRequired,
    snack: _.shape({
        open: _.bool.isRequired,
        message: _.string,
        variant: _.oneOf(['error', 'success']),
    }).isRequired,
    onHideNotification: _.func.isRequired,
}

const mapDispatchToProps = dispatch => ({
    onHideNotification: () => {
        dispatch(hideNotification())
    }
})

const mapStateToProps = ({ notification }) => ({
    snack: notification.snack,
})

export default connect(
    mapStateToProps,
    mapDispatchToProps,
)(LayoutContainer)
