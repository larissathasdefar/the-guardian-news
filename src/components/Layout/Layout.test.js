import React from 'react'
import { render } from '@testing-library/react'
import Layout from './Layout'

describe('Layout component', () => {
    test('Renders Layout component', async () => {
        const { findByTestId } = render(
            <Layout onHideNotification={() => {}}>
                Testando o componente de layout do projeto
            </Layout>
        )
        const component = await findByTestId('layout-content')

        expect(component).toHaveTextContent('The Guardian')
        expect(component).toHaveTextContent('Testando o componente de layout do projeto')
    })

    test('Snapshot', () => {
        const { asFragment } = render(
            <Layout onHideNotification={() => {}}>
                Testando o componente de layout do projeto
            </Layout>
        )
        expect(asFragment()).toMatchSnapshot()
    })

})
