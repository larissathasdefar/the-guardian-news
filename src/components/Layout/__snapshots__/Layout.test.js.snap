// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout component Snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-AxirZ cjqdBc"
    data-testid="layout-content"
  >
    <div
      class="sc-AxiKw cHBOJS"
    >
      <h3
        class="MuiTypography-root MuiTypography-h3"
      >
        The Guardian
      </h3>
    </div>
    <div
      class="sc-AxhCb cLmHhN"
    >
      Testando o componente de layout do projeto
    </div>
  </div>
</DocumentFragment>
`;
